.github/workflows/python-package-genai.yml

```diff
@@ -39,7 +39,7 @@ jobs:
       fail-fast: false
```

Comment:
In the future, could we put all formatting-related changes in their own commit or even better in their own PR?

Reply:
What do you mean? Usually, pre-commit is turned on, so the formatting would happen for every commit. Since it was turned off as part of this development work, re-enabling it immediately ran all the hooks. And then it warned me about types, other static errors, etc., so I fixed those manually, but it wouldn't let me commit without fixing everything.

I suppose I could have let the auto-formatting run, then uninstalled pre-commit, pushed a commit, re-installed pre-commit, and made any manual changes in a separate commit if that's in line with what you're asking for?

Reply:
I'm not sure how to achieve it, but I mean literally what I said. I had to read everything way closer because I couldn't tell if it was going to be a formatting related change or some other kind of change (e.g. using a python 3.10 feature or something).

It would be much easier to review if a single commit contained anything related to auto-formatting. Like run precommit without any feature-related changes for the first commit, then future commits are the non-formatting changes, like adding a trialing comma or whatever.

Reply:
Totally fair. I had originally had it as a separate commit and will do so next time.

Just a heads up that I didn't make the aforementioned change. The trialing comma was actually pre-commit formatting. The only things I did were fixing the errors pre-commit returned, then fixing the 2-3 CLI unit tests that broke when the behavior was fixed. I think you're right though - I can run it just with auto-format and save that as a commit so that you don't need to review that commit.

Thanks again for reviewing, Matt!